### DISCUSSION TITLE
can't both create an output folder and get cache hits

### DISCUSSION THREAD

**Discussion by BrianHicks** [vercel/turborepo#6930] [Help]
I'm trying to figure out a good pattern for packaging up some SCSS using Turborepo. Here's what I'd like to do:

- create a `build/stylesheets` directory
- copy some vendored files from my `node_modules` into it
- run `sass` to compile my stylesheet into it
- after that is all done, I want to use the `postcss` CLI to do some post-processing for production into a `dist` directory

Here's the way I have the task graph laid out via `dependsOn` relationships:

```mermaid
flowchart TD
    build["build:styles:dir"]
    copy["build:styles:vendor"] --> build
    sass["build:styles:compile"] --> build
    prod["dist:styles"] --> copy
    prod --> sass
```

This works fine, except that the step to create the build directory is inheriting too many dependencies. If I don't specify any, it gets all the files in the workspace. If I specify an empty list specifically, it seems to have the same behavior. (In fact, the output of `turbo run build:styles:dir --summarize` confirms that the empty list isn't respected.) This means that despite carefully specifying all the dependencies of the build steps, I get cache misses as soon as I change any files, even those unrelated to styling.

So here's my question: is there a better way to do this? I suppose I could have each step do a `mkdir -p build/stylesheets`, but that seems like a silly repetition when I have a nice build graph to stick it in *right there*. Or maybe is this "explicitly empty inputs means all inputs" a bug? I'm really not sure!

**Comment by mehulkar** [comment]
>  except that the step to create the build directory is inheriting too many dependencies. If I don't specify any, it gets all the files in the workspace. If I specify an empty list specifically, it seems to have the same behavior. 

Can you show the config you're using here? `inputs: []` shoullld work, but I'm not sure.

**Reply by BrianHicks** [reply]
I ended up breaking this up into a bunch of separate packages to fit an organizational best practice. It's certainly possible I was configuring something incorrectly.
